File: B3WM.Flutter/lib/models/verifier_state.dart
import 'signal_event.dart';
import 'verifier_config.dart';
import 'verifier_pending_signal.dart';
import 'verifier_position.dart';

class VerifierState {
  String symbol;
  int timeFrame;
  bool isRunning;
  VerifierConfig? config;
  VerifierPosition? openPosition;
  VerifierPendingSignal? pendingSignal;
  int totalTrades;
  int winCount;
  int lossCount;
  double winRate;
  double netProfit;
  double grossProfit;
  double grossLoss;
  double maxDrawdown;
  List<SignalEvent> signals;
  List<double> equityCurve;

  VerifierState({
    required this.symbol,
    required this.timeFrame,
    this.isRunning = false,
    this.config,
    this.openPosition,
    this.pendingSignal,
    this.totalTrades = 0,
    this.winCount = 0,
    this.lossCount = 0,
    this.winRate = 0,
    this.netProfit = 0,
    this.grossProfit = 0,
    this.grossLoss = 0,
    this.maxDrawdown = 0,
    List<SignalEvent>? signals,
    List<double>? equityCurve,
  })  : signals = signals ?? [],
        equityCurve = equityCurve ?? [];

  factory VerifierState.fromJson(Map<String, dynamic> json) {
    return VerifierState(
      symbol: json['symbol'] as String? ?? '',
      timeFrame: (json['timeFrame'] as num?)?.toInt() ?? 0,
      isRunning: json['isRunning'] as bool? ?? false,
      config: json['config'] != null
          ? VerifierConfig.fromJson(json['config'] as Map<String, dynamic>)
          : null,
      openPosition: json['openPosition'] != null
          ? VerifierPosition.fromJson(
              json['openPosition'] as Map<String, dynamic>)
          : null,
      pendingSignal: json['pendingSignal'] != null
          ? VerifierPendingSignal.fromJson(
              json['pendingSignal'] as Map<String, dynamic>)
          : null,
      totalTrades: (json['totalTrades'] as num?)?.toInt() ?? 0,
      winCount: (json['winCount'] as num?)?.toInt() ?? 0,
      lossCount: (json['lossCount'] as num?)?.toInt() ?? 0,
      winRate: (json['winRate'] as num?)?.toDouble() ?? 0,
      netProfit: (json['netProfit'] as num?)?.toDouble() ?? 0,
      grossProfit: (json['grossProfit'] as num?)?.toDouble() ?? 0,
      grossLoss: (json['grossLoss'] as num?)?.toDouble() ?? 0,
      maxDrawdown: (json['maxDrawdown'] as num?)?.toDouble() ?? 0,
      signals: (json['signals'] as List?)
              ?.map((e) => SignalEvent.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
      equityCurve: (json['equityCurve'] as List?)
              ?.map((e) => (e as num).toDouble())
              .toList() ??
          [],
    );
  }
}

File: B3WM.Shared/Models/VerifierState.cs
using B3WM.Shared.Models.Backtest;

namespace B3WM.Shared.Models
{
    public class VerifierPosition
    {
        public OrderSide Side { get; set; }
        public double EntryPrice { get; set; }
        public double StopPrice { get; set; }
        public double TargetPrice { get; set; }
        public int Quantity { get; set; }
        public DateTime EntryDate { get; set; }
        public string? EntryReason { get; set; }
    }

    public class VerifierPendingSignal
    {
        public OrderSide Side { get; set; }
        public string? Reason { get; set; }
        public double StopLossPrice { get; set; }
        public double TakeProfitPrice { get; set; }
    }

    public class VerifierState
    {
        public string Symbol { get; set; } = string.Empty;
        public int TimeFrame { get; set; }
        public bool IsRunning { get; set; }
        public VerifierConfig? Config { get; set; }
        public VerifierPosition? OpenPosition { get; set; }
        public VerifierPendingSignal? PendingSignal { get; set; }
        public int TotalTrades { get; set; }
        public int WinCount { get; set; }
        public int LossCount { get; set; }
        public double WinRate { get; set; }
        public double NetProfit { get; set; }
        public double GrossProfit { get; set; }
        public double GrossLoss { get; set; }
        public double MaxDrawdown { get; set; }
        public List<SignalEvent> Signals { get; set; } = new();
        public List<double> EquityCurve { get; set; } = new();
    }
}
